### ISSUE TITLE
Add compatibility to Authentik (+ Hass-Auth-Header)

### ISSUE THREAD

**Issue by chriexpe** [home-assistant/android#4015]
I use Home Assistant with NGINX + Authentik + this integration https://github.com/BeryJu/hass-auth-header so that I don't have to login two times, on the phone and web browser everything works just fine, but when I try to login on my Watch4, on that "login into Wear OS Device" screen I always get this error: "It was not possible to register the watch", and the only way to get it working is by removing Authentik.
I couldn't find any logs about this error on Home Assistant.

**Comment by dshokouhi** [comment]
duplicate of https://github.com/home-assistant/android/issues/1438



> I couldn't find any logs about this error on Home Assistant.

Sometimes filtering by `homeassistant` is not good enough, the error may come from somewhere else

**Comment by chriexpe** [comment]
Wait, but I'm referring to connecting the WearOS Home Assistant to my instance!
This is the log:
11-16 21:34:34.755 12013  6595 D WearOnboardingListener: onMessageReceived: MessageEventParcelable[17255,/request_home_assistant_instance, size=0]
11-16 21:34:34.755 12013  6595 D WearOnboardingListener: sendHomeAssistantInstance: 4121a1b3
11-16 21:34:34.756 12013  6595 D ServerConnectionInfo: Using external URL
11-16 21:34:34.769 12013 12013 D WearOnboardingListener: sendHomeAssistantInstance: success
11-16 21:34:35.505 12013 12013 D IntegrationRepository: setAppActive(): false
11-16 21:34:35.506 12013 12013 D IntegrationRepository: setAppActive(): false
11-16 21:34:35.542 12013 12013 D SettingsWearAct: Waiting on Results for both connected nodes and nodes with app
11-16 21:34:35.547 12013 12013 D WindowManager: Add to mViews: DecorView@53dd76[SettingsWearActivity],pkg= io.homeassistant.companion.android
11-16 21:34:35.554 12013 12013 I Quality : Skipped: false 2 cost 23.520325 refreshRate 0 processName io.homeassistant.companion.android
11-16 21:34:35.568 12013 12013 V OplusZoomWindowDecorViewHelper: setLastReportedMergedConfiguration mZoomDisplayHeight: 3120 getDecorView.5496182
11-16 21:34:35.574 12013 12013 D SettingsWearAct: Waiting on Results for both connected nodes and nodes with app
11-16 21:34:35.575 12013  8632 D SurfaceComposerClient: VRR [FRTC] client handle [bufferId:18446744073709551615 framenumber:0] [ffffffff, ffffffff]
11-16 21:34:35.575 12013 12013 D SettingsWearAct: Capable Nodes: [Node{Galaxy Watch4 (Y43X), id=4121a1b3, hops=1, isNearby=true}]
11-16 21:34:35.575 12013 12013 D SettingsWearAct: Installed on all devices
11-16 21:34:35.607 12013 12013 I Quality : Skipped: false 3 cost 43.093803 refreshRate 0 processName io.homeassistant.companion.android
11-16 21:34:35.647 12013 12013 D WindowManager: Add to mViews: DecorView@5a1d33f[SettingsWearMainView],pkg= io.homeassistant.companion.android
11-16 21:34:35.653 12013 12013 I Quality : Skipped: false 1 cost 11.575034 refreshRate 0 processName io.homeassistant.companion.android
11-16 21:34:35.676 12013 12013 V OplusZoomWindowDecorViewHelper: setLastReportedMergedConfiguration mZoomDisplayHeight: 3120 getDecorView.94491455
11-16 21:34:35.683 12013 12013 D SettingsWearViewModel: Requesting config from node 4121a1b3
11-16 21:34:35.685 12013 12013 I Quality : Skipped: false 1 cost 21.27904 refreshRate 0 processName io.homeassistant.companion.android
11-16 21:34:35.686 12013 12034 D SurfaceComposerClient: VRR [FRTC] client handle [bufferId:18446744073709551615 framenumber:0] [ffffffff, ffffffff]
11-16 21:34:35.713 12013 12013 D SettingsWearViewModel: Request for config sent successfully
11-16 21:34:35.715 12013 12013 I Quality : Skipped: false 1 cost 17.315556 refreshRate 0 processName io.homeassistant.companion.android
11-16 21:34:35.926 12013 12013 D SettingsWearViewModel: onDataChanged 1
11-16 21:34:35.968 12013 12013 D WindowManager: Add to mViews: DecorView@57fde7e[OnboardingActivity],pkg= io.homeassistant.companion.android
11-16 21:34:35.991 12013 12013 W cr_SupportWebSettings: setForceDarkBehavior() is a no-op in an app with targetSdkVersion>=T
11-16 21:34:35.991 12013 12013 W cr_WebSettings: setForceDark() is a no-op in an app with targetSdkVersion>=T
11-16 21:34:36.010 12013 12013 V OplusZoomWindowDecorViewHelper: setLastReportedMergedConfiguration mZoomDisplayHeight: 3120 getDecorView.92266110
11-16 21:34:36.021 12013 12553 W chromium: [WARNING:display_webview.cc(32)] WebView overlays are enabled!
11-16 21:34:36.032 12013  8632 D SurfaceComposerClient: VRR [FRTC] client handle [bufferId:18446744073709551615 framenumber:0] [ffffffff, ffffffff]
11-16 21:34:36.035 12013 12013 I Quality : Skipped: false 5 cost 59.44853 refreshRate 0 processName io.homeassistant.companion.android
11-16 21:34:36.446 12013 12485 E BLASTBufferQueue: BLASTBufferItemConsumer::onDisconnect()
11-16 21:34:36.446 12013 12485 D OpenGLRenderer: endAllActiveAnimators on 0xb400007a0fc96c00 (RippleDrawable) with handle 0xb400007aa6b35aa0
11-16 21:34:36.446 12013 12485 D OpenGLRenderer: endAllActiveAnimators on 0xb400007a22ae6000 (RippleDrawable) with handle 0xb400007aa6b35e60
11-16 21:34:36.446 12013 12485 D OpenGLRenderer: endAllActiveAnimators on 0xb400007a2fc35c00 (RippleDrawable) with handle 0xb400007aa6b35580
11-16 21:34:36.446 12013 12485 D OpenGLRenderer: endAllActiveAnimators on 0xb400007a16077400 (RippleDrawable) with handle 0xb400007aa6b35f00
11-16 21:34:36.446 12013 12485 D OpenGLRenderer: endAllActiveAnimators on 0xb400007a1607b800 (RippleDrawable) with handle 0xb400007aa6b35460
11-16 21:34:36.446 12013 12485 D OpenGLRenderer: endAllActiveAnimators on 0xb400007a1607ca00 (RippleDrawable) with handle 0xb400007aa6b5b180
11-16 21:34:36.446 12013 12485 D OpenGLRenderer: endAllActiveAnimators on 0xb400007a22a50c00 (RippleDrawable) with handle 0xb400007aa6bbe0e0
11-16 21:34:36.458 12013 12013 D OplusScrollToTopManager: unregisterGuidePopupDismissReceiverInSystemUI
11-16 21:34:36.458 12013 12013 D OplusScrollToTopManager: Receiver not registered: android.view.OplusScrollToTopManager$1@f4a575f
11-16 21:34:36.458 12013 12013 D OplusScrollToTopManager: unregisterSystemUIBroadcastReceiver 
11-16 21:34:36.458 12013 12013 D OplusScrollToTopManager: java.lang.IllegalArgumentException: Receiver not registered: android.view.OplusScrollToTopManager$2@bc055ac
11-16 21:34:36.458 12013 12013 V OplusZoomWindowDecorViewHelper: removeZoomView
11-16 21:34:36.459 12013 12485 E BLASTBufferQueue: BLASTBufferItemConsumer::onDisconnect()
11-16 21:34:36.466 12013 12013 I Quality : Skipped: false 1 cost 11.633463 refreshRate 0 processName io.homeassistant.companion.android
11-16 21:34:36.468 12013 12485 E BLASTBufferQueue: BLASTBufferItemConsumer::onDisconnect()
11-16 21:34:36.571 12013 12013 I Quality : Skipped: false 1 cost 16.211723 refreshRate 0 processName io.homeassistant.companion.android
11-16 21:34:37.397 12013 12013 D SettingsWearViewModel: Successfully sent auth to wear
11-16 21:34:37.398 12013 12013 D SettingsWearViewModel: onDataChanged 1
11-16 21:34:37.400 12013  8632 D SurfaceComposerClient: VRR [FRTC] client handle [bufferId:18446744073709551615 framenumber:0] [ffffffff, ffffffff]
11-16 21:34:37.816 12013 12485 E BLASTBufferQueue: BLASTBufferItemConsumer::onDisconnect()
11-16 21:34:37.817 12013 12485 D OpenGLRenderer: endAllActiveAnimators on 0xb400007a16171e00 (UnprojectedRipple) with handle 0xb400007a994e9860
11-16 21:34:37.847 12013 12013 D OplusScrollToTopManager: unregisterGuidePopupDismissReceiverInSystemUI
11-16 21:34:37.847 12013 12013 D OplusScrollToTopManager: Receiver not registered: android.view.OplusScrollToTopManager$1@e833ccd
11-16 21:34:37.847 12013 12013 D OplusScrollToTopManager: unregisterSystemUIBroadcastReceiver 
11-16 21:34:37.847 12013 12013 D OplusScrollToTopManager: java.lang.IllegalArgumentException: Receiver not registered: android.view.OplusScrollToTopManager$2@df58982
11-16 21:34:37.847 12013 12013 V OplusZoomWindowDecorViewHelper: removeZoomView
11-16 21:34:37.897 12013  6618 D WearOnboardingListener: onMessageReceived: MessageEventParcelable[17257,/request_home_assistant_instance, size=0]
11-16 21:34:37.897 12013  6618 D WearOnboardingListener: sendHomeAssistantInstance: 4121a1b3
11-16 21:34:37.898 12013  6618 D ServerConnectionInfo: Using external URL
11-16 21:34:37.905 12013 12013 D WearOnboardingListener: sendHomeAssistantInstance: success
11-16 21:34:47.011 12013 12013 W Choreographer: Frame time is 0.008618 ms in the future!  Check that graphics HAL is generating vsync timestamps using the correct timebase.
11-16 21:35:01.530 12013 12013 W Choreographer: Frame time is 0.075471 ms in the future!  Check that graphics HAL is generating vsync timestamps using the correct timebase.
11-16 21:35:05.539 12013 12013 W Choreographer: Frame time is 0.072004 ms in the future!  Check that graphics HAL is generating vsync timestamps using the correct timebase.
11-16 21:35:07.650 12013 12013 D SettingsWearViewModel: onDataChanged 1
11-16 21:35:07.652 12013 12013 E SettingsWearViewModel: Watch was unable to register: java.io.InterruptedIOException: timeout
11-16 21:35:07.652 12013 12013 E SettingsWearViewModel: 	at okhttp3.internal.connection.RealCall.timeoutExit(RealCall.kt:398)
11-16 21:35:07.652 12013 12013 E SettingsWearViewModel: 	at okhttp3.internal.connection.RealCall.callDone(RealCall.kt:360)
11-16 21:35:07.652 12013 12013 E SettingsWearViewModel: 	at okhttp3.internal.connection.RealCall.noMoreExchanges$okhttp(RealCall.kt:325)
11-16 21:35:07.652 12013 12013 E SettingsWearViewModel: 	at okhttp3.internal.connection.RealCall.getResponseWithInterceptorChain$okhttp(RealCall.kt:209)
11-16 21:35:07.652 12013 12013 E SettingsWearViewModel: 	at okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:517)
11-16 21:35:07.652 12013 12013 E SettingsWearViewModel: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1137)
11-16 21:35:07.652 12013 12013 E SettingsWearViewModel: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:637)
11-16 21:35:07.652 12013 12013 E SettingsWearViewModel: 	at java.lang.Thread.run(Thread.java:1012)
11-16 21:35:07.652 12013 12013 E SettingsWearViewModel: Caused by: java.io.IOException: Canceled
11-16 21:35:07.652 12013 12013 E SettingsWearViewModel: 	at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.kt:72)
11-16 21:35:07.652 12013 12013 E SettingsWearViewModel: 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
11-16 21:35:07.652 12013 12013 E SettingsWearViewModel: 	at okhttp3.internal.connection.RealCall.getResponseWithInterceptorChain$okhttp(RealCall.kt:201)
11-16 21:35:07.652 12013 12013 E SettingsWearViewModel: 	... 4 more
11-16 21:35:07.652 12013 12013 D SettingsWearViewModel: onDataChanged 1
11-16 21:35:07.690 12013 12013 I Quality : Skipped: false 2 cost 24.630913 refreshRate 0 processName io.homeassistant.companion.android
11-16 21:35:10.786 12013 12013 D Activity: io.homeassistant.companion.android.settings.wear.views.SettingsWearMainView@2dc2844onBackPressed!
11-16 21:35:10.796 12013 12013 I Quality : Skipped: false 1 cost 14.284139 refreshRate 0 processName io.homeassistant.companion.android
11-16 21:35:10.829 12013 12013 D IntegrationRepository: isAppLocked(): false. (LockEnabled: false, appActive: false, expireMillis: 1700181276505, currentMillis: 1700181310829)
11-16 21:35:10.905 12013 30215 D SurfaceComposerClient: VRR [FRTC] client handle [bufferId:18446744073709551615 framenumber:0] [ffffffff, ffffffff]
11-16 21:35:10.906 12013 12013 I Quality : Skipped: false 6 cost 67.68547 refreshRate 0 processName io.homeassistant.companion.android
11-16 21:35:10.943 12013 12013 D IntegrationRepository: isAppLocked(): false. (LockEnabled: false, appActive: false, expireMillis: 1700181276505, currentMillis: 1700181310943)
11-16 21:35:10.943 12013 12013 D IntegrationRepository: setAppActive(): true
11-16 21:35:11.335 12013 12485 E BLASTBufferQueue: BLASTBufferItemConsumer::onDisconnect()
11-16 21:35:11.335 12013 12485 D OpenGLRenderer: endAllActiveAnimators on 0xb400007a1608bc00 (UnprojectedRipple) with handle 0xb400007a98b66c40
11-16 21:35:11.355 12013 12013 D OplusScrollToTopManager: unregisterGuidePopupDismissReceiverInSystemUI
11-16 21:35:11.356 12013 12013 D OplusScrollToTopManager: Receiver not registered: android.view.OplusScrollToTopManager$1@568ccaa
11-16 21:35:11.356 12013 12013 D OplusScrollToTopManager: unregisterSystemUIBroadcastReceiver 
11-16 21:35:11.356 12013 12013 D OplusScrollToTopManager: java.lang.IllegalArgumentException: Receiver not registered: android.view.OplusScrollToTopManager$2@177209b
11-16 21:35:11.356 12013 12013 V OplusZoomWindowDecorViewHelper: removeZoomView
11-16 21:35:11.366 12013 12013 I Quality : Skipped: false 2 cost 27.34949 refreshRate 0 processName io.homeassistant.companion.android

**Comment by dshokouhi** [comment]
> 11-16 21:35:07.652 12013 12013 E SettingsWearViewModel: Watch was unable to register: java.io.InterruptedIOException: timeout

this error is still not one the app can fix, it might be a configuration issue I am not sure but we cannot fix networking issues like a timeout error, those are for the user to fix.

**Comment by chriexpe** [comment]
> > 11-16 21:35:07.652 12013 12013 E SettingsWearViewModel: Watch was unable to register: java.io.InterruptedIOException: timeout
> 
> this error is still not one the app can fix, it might be a configuration issue I am not sure but we cannot fix networking issues like a timeout error, those are for the user to fix.

oh got it, but just by any chance, do you know what Home Assistant is requesting when it gets timeout? Or maybe as it's a different device the token used on the phone (from authentik) doesn't work on the watch.
